core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -1640,6 +1640,18 @@ class LogTest {
     assertEquals(partition.toInt, topicPartition.partition)
   }
 
+  // see https://issues.apache.org/jira/browse/KAFKA-5232
+  @Test
+  def testParseTopicPartitionNameForDeletedTopic() {
+    val topic = "foo.bar-testtopic";
+    val partition = "42";
+    val deleteMarkerSuffix = "." + java.util.UUID.randomUUID.toString.replaceAll("-","") + "-delete"
+    val dir = new File(logDir + topicPartitionName(topic, partition) + deleteMarkerSuffix)
+    val topicPartition = Log.parseTopicPartitionName(dir)
+    assertEquals("Unexpected topic name parsed", topic, topicPartition.topic)
+    assertEquals("Unexpected partition number parsed", partition.toInt, topicPartition.partition)
+  }
```

Comment:
While we're doing this, it would be better to flesh out tests for the deleted case. Namely variants of `testParseTopicPartitionNameForMissingSeparator`, `testParseTopicPartitionNameForMissingTopic` and `testParseTopicPartitionNameForMissingPartition` with the `-delete` suffix.